internal/service/currency_service.go

```diff
@@ -29,56 +30,44 @@ var usdCode = getCurrencyCode("USD")
 var uahCode = getCurrencyCode("UAH")
 
 type CurrencyService struct {
-	// Cache variables
-	cacheRate  float64
 	cacheTime  time.Time
+	repo       *repository.ExchangeRateRepository
```

Comment:
it's common to use interfaces for dependencies, because it's easier to test later.